Sources/PackagePlugin/Protocols.swift

```diff
@@ -81,3 +81,21 @@ extension BuildToolPlugin {
     }
 }
 
+/// Defines functionality for all plugins that have a `customCommand` capability.
+public protocol CommandPlugin: Plugin {
+    /// Invoked by SwiftPM to perform the custom actions of the command.
+    func performCommand(
+        /// The context in which the plugin is invoked. This is the same for all
+        /// kinds of plugins, and provides access to the package graph, to cache
+        /// directories, etc.
+        context: PluginContext,
+        
+        /// The targets to which the command should be applied. If the invoker of
+        /// the command has not specified particular targets, this will be a list
+        /// of all the targets in the package to which the command is applied.
+        targets: [Target],
+        
+        /// Any literal arguments passed after the verb in the command invocation.
+        arguments: [String]
+    ) throws
```

Comment:
should this be async?

Reply:
Yes definitely, but there are currently issues with the back deployment.  I have a separate PR that adds async to all the plugin APIs, and plan to do it there.